src/org/opendatakit/briefcase/ui/ScrollingStatusListDialog.java

```diff
@@ -128,8 +130,12 @@ private ScrollingStatusListDialog(Frame frame, String labelText, String title, I
   public void onEvent(FormStatusEvent event) {
     // Since there can be multiple FormStatusEvent's published concurrently,
     // we have to check if the event is meant for this dialog instance.
-    if (event.getStatus().getFormDefinition().equals(form)) {
-      editorArea.setText(event.getStatus().getStatusHistory());
+    if (isShowing() && event.getStatus().getFormDefinition().equals(form)) {
+      try {
+        editorArea.read(new StringReader(event.getStatus().getStatusHistory()), null);
```

Comment:
So, this should only be an improvement on the previous code in terms of semantics. It also was done based on actual profiling of the allocations. It's effectively setText with the entire history just as before, but history is bounded so allocations don't grow linearly.

Careful with that suggestion - you just made the JEditorPane's buffer O(n) for memory consumption. I didn't attempt to gold-plate the existing solution, only fix the scale issue. If I were to spend any more time on this myself, my first attempt would be to use the logging framework and make these views simply bounded buffers that are polled periodically, not on every status updated.